bot/utils/persist.py

```diff
@@ -0,0 +1,40 @@
+import sqlite3
+from pathlib import Path
+from shutil import copyfile
+
+DIRECTORY = Path("data")  # directory that has a persistent volume mapped to it
+
+
+def datafile(file_path: Path) -> Path:
+    """
+    Copy datafile at the provided file_path to the persistent data directory.
+
+    A persistent data file is needed by some features in order to not lose data
+    after bot rebuilds.
+
+    This function will ensure that a clean data file with default schema,
+    structure or data is copied over to the persistent volume before returning
+    the path to this new persistent version of the file.
+
+    If the persistent file already exists, it won't be overwritten with the
+    clean default file, just returning the Path instead to the existing file.
+
+    Example Usage:
+    >>> clean_default_datafile = Path("bot", "resources", "datafile.json")
+    >>> persistent_file_path = datafile(clean_default_datafile)
+    """
+    if not file_path.exists():
```

Comment:
Probably a good idea to check it's a file and not a directory... just in case. It also doubles as check for if the file exists. It may be worth revising the exception if this is changed though.
```suggestion
    if not file_path.is_file():
```